.github/workflows/mac.yml

```diff
@@ -0,0 +1,103 @@
+name: Mac_CI
+
+on:
+  push:
+    branches:
+    - main
+    - rel-*
+  pull_request:
+    branches:
+    - main
+    - rel-*
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+env:
+  python_version: 3.11
+  xcode_version: 15.2
+
+jobs:
+  ARM64:
+    runs-on: macos-14
+
+    timeout-minutes: 60
+
+    steps:
+    - uses: actions/setup-python@v5
+      with:
+        python-version: ${{ env.python_version }}
+
+    - name: Verify ARM64 machine
+      shell: python
+      run: |
+        import platform
+        assert platform.machine() == "arm64", "This job expects to be run on an ARM64 machine."
+
+    - uses: actions/checkout@v4
+
+    - name: Use Xcode ${{ env.xcode_version }}
+      shell: bash
+      run: |
+        XCODE_DEVELOPER_DIR="/Applications/Xcode_${{ env.xcode_version }}.app/Contents/Developer"
+        sudo xcode-select --switch "${XCODE_DEVELOPER_DIR}"
+
+    - name: Build and test
+      shell: bash
+      run: |
+        python ./tools/ci_build/build.py \
+          --build_dir ./build \
+          --update \
+          --build --parallel \
+          --test \
+          --build_shared_lib \
+          --build_objc \
+          --use_xnnpack \
+          --use_binskim_compliant_compile_flags
+
+        # TODO add --use_coreml once unit test failures are addressed

```

Comment:
unit test failures with CoreML EP enabled:
```
2: [  FAILED  ] 3 tests, listed below:
2: [  FAILED  ] MathOpTest.Add_Broadcast_0x0
2: [  FAILED  ] MathOpTest.MatMulFloatType
2: [  FAILED  ] MathOpTest.MatMulFloatTypeInitializer
```

https://github.com/microsoft/onnxruntime/actions/runs/9135671431/job/25123372530

Reply:
will address in a future PR